test/framework/ingress.go

```diff
@@ -23,238 +24,165 @@ func (s HostAndPort) Port() string {
-	require.NoError(t, err)
+	f.deployIngress(t, k8sContext, namespace, "reaper-ingress.yaml", "reaper", reaperTemplate,
+		&ingressKustomization{reaperServiceName, reaperHostAndPort.Host()})
 }
 
 func (f *E2eFramework) DeploySolrIngresses(t *testing.T, k8sContext, namespace, solrServiceName string, solrHostAndPort HostAndPort) {
-	src := filepath.Join("..", "..", "test", "testdata", "ingress", "solr-ingress.yaml")
-	dest := filepath.Join("..", "..", "build", "test-config", "ingress", "solr", k8sContext)
-	_, err := utils.CopyFileToDir(src, dest)
-	require.NoError(t, err)
-	err = generateSolrIngressKustomization(k8sContext, namespace, solrServiceName, solrHostAndPort.Host())
-	require.NoError(t, err)
-	err = f.kustomizeAndApply(dest, namespace, k8sContext)
-	require.NoError(t, err)
+	f.deployIngress(t, k8sContext, namespace, "solr-ingress.yaml", "solr", solrTemplate,
+		&ingressKustomization{solrServiceName, solrHostAndPort.Host()})
 }
 
 func (f *E2eFramework) DeployGraphIngresses(t *testing.T, k8sContext, namespace, graphServiceName string, graphHostAndPort HostAndPort) {
-	src := filepath.Join("..", "..", "test", "testdata", "ingress", "graph-ingress.yaml")
-	dest := filepath.Join("..", "..", "build", "test-config", "ingress", "graph", k8sContext)
+	f.deployIngress(t, k8sContext, namespace, "graph-ingress.yaml", "graph", graphTemplate,
+		&ingressKustomization{graphServiceName, graphHostAndPort.Host()})
+}
+
+func (f *E2eFramework) deployIngress(t *testing.T, k8sContext, namespace, sourceYaml, buildDir, template string, templateData interface{}) {
+	src := filepath.Join("..", "..", "test", "testdata", "ingress", sourceYaml)
+	dest := filepath.Join("..", "..", "build", "test-config", "ingress", buildDir, k8sContext)
 	_, err := utils.CopyFileToDir(src, dest)
 	require.NoError(t, err)
-	err = generateGraphIngressKustomization(k8sContext, namespace, graphServiceName, graphHostAndPort.Host())
+
+	err = generateKustomizationFile(fmt.Sprintf("ingress/%s/%s", buildDir, k8sContext), templateData, template)
 	require.NoError(t, err)
+
 	err = f.kustomizeAndApply(dest, namespace, k8sContext)
 	require.NoError(t, err)
 }
 
 func (f *E2eFramework) UndeployAllIngresses(t *testing.T, k8sContext, namespace string) {
 	options := kubectl.Options{Context: k8sContext, Namespace: namespace}
-	err := kubectl.DeleteAllOf(options, "IngressRoute")
-	assert.NoError(t, err)
-	err = kubectl.DeleteAllOf(options, "IngressRouteTCP")
+	err := kubectl.DeleteAllOf(options, "Ingress")
 	assert.NoError(t, err)
 }
 
-type ingressKustomization struct {
-	Namespace   string
-	ServiceName string
-	Host        string
-}
-
-func generateStargateIngressKustomization(k8sContext, namespace, serviceName, host string) error {
-	tmpl := `apiVersion: kustomize.config.k8s.io/v1beta1
+const stargateTemplate = `apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 resources:
 - stargate-ingress.yaml
-namespace: {{ .Namespace }}
```

Comment:
I removed the namespace from the YAML templates.
- it's one less thing to handle in the Kustomization
- it avoids the possible confusion for CQL, where there are two different ones: ConfigMap namespace and Service namespace.
- the one eventually passed to `kubectl apply` is the only one that matters anyway (the command would fail if it didn't match).

Reply:
Kustomize does not care about ``--namespace``  given to kubectl, only the templates and what you have there matters. The value from kubectl does not swim to the template rendering (and without namespace in templates, the namespace transformers might not work correctly)

Reply:
What I was aiming for is no namespace transformation at all. In [this code snippet]( https://github.com/k8ssandra/k8ssandra-operator/blob/main/test/framework/e2e_framework.go#L273-L279):
```go
	buf, err := kustomize.BuildDir(kdir)
        [...]
	options := kubectl.Options{Context: context, ServerSide: true, Namespace: namespace}
	if err := kubectl.Apply(options, buf); err != nil {
		return err
	}
```
`kustomize.BuildDir` takes a template that has no namespace (e.g. `stargate-ingress.yaml`), and transforms it into a YAML that has no namespace either. `kubectl.Apply` will create the resource in the correct namespace anyway, so do we need it in the generated YAML file?